game/hiscore.cc

```diff
@@ -120,17 +110,27 @@ void Hiscore::load(xmlpp::NodeSet const& nodes) {
 }
 
 void Hiscore::save(xmlpp::Element *hiscores) {
-	for (auto const& h: m_hiscore) {
-		xmlpp::Element* hiscore = xmlpp::add_child_element(hiscores, "hiscore");
-		hiscore->set_attribute("playerid", std::to_string(h.playerid));
-		hiscore->set_attribute("songid", std::to_string(h.songid));
-		hiscore->set_attribute("track", h.track);
-		hiscore->set_attribute("level", std::to_string(h.level));
-		hiscore->set_attribute("unixtime", std::to_string(h.unixtime.count()));
-		hiscore->add_child_text(std::to_string(h.score));
+	for (auto const& [song_id, song_hiscores] : m_hiscore_map) {
```

Comment:
we could sort by timestamp, song and/or player id before saving.
As the map is unordered entries in the db will be shuffled (stable order, just not by song or player id) right now.